### PR TITLE
json-ld cached doc loader

### PR DESCRIPTION
**Pull request by singyiu**
[json-gold](https://github.com/piprate/json-gold) does have the caching document loader implemented (RFC7324)
It is enabled by passing the caching loader into the ldProcessor options
Added it as the default option [here](https://github.com/singyiu/ssi-sdk/blob/c1f4324e9d365481dde54b47af76be47e0f82b67/util/helpers.go#L44).

Json-ld context can be loaded as a json-ld document (type map[string]interface{}) using the loadDocument function [test example](https://github.com/singyiu/ssi-sdk/blob/c1f4324e9d365481dde54b47af76be47e0f82b67/util/helpers_test.go#L115)
or further converting it into a json-gold.Context object [test example](https://github.com/singyiu/ssi-sdk/blob/c1f4324e9d365481dde54b47af76be47e0f82b67/util/helpers_test.go#L134)
A helper function "GetContextFromMap" was added.

### CODE REVIEW

File: util/helpers_test.go [hunk 1]
```diff
@@ -90,3 +92,48 @@ func TestMergeUniqueValues(t *testing.T) {
 		assert.True(tt, len(res) == len(a)+2)
 	})
 }
+
+func TestLDProcessor(t *testing.T) {
+	testJsonDLContextUrlStr := "http://schema.org/"
```
Comment (decentralgabe) [inline util/helpers_test.go:97]: nit: `testJSONLDContextURLStr`

File: util/helpers_test.go [hunk 2]
```diff
@@ -90,3 +92,48 @@ func TestMergeUniqueValues(t *testing.T) {
...
+		contextMap := map[string]interface{}{
+			"dc": "http://purl.org/dc/elements/1.1/",
+			"ex": "http://example.org/vocab#",
+			"ex:contains": map[string]interface{}{
+				"@type": "@id",
+			},
+		}
+
+		activeCtx, err := ldProcessor.GetContextFromMap(contextMap)
+		//activeCtx: &{values:map[@base: processingMode:json-ld-1.1] options:0xc0001288f0 termDefinitions:map[dc:map[@id:http://purl.org/dc/elements/1.1/ @reverse:false _prefix:true] ex:map[@id:http://example.org/vocab# @reverse:false _prefix:true] ex:contains:map[@id:http://example.org/vocab#contains @reverse:false @type:@id]] inverse:map[] protected:map[] previousContext:<nil>}
```
Comment (decentralgabe) [inline util/helpers_test.go:135]: perhaps a note calling out that this is the expected result?

**Review by decentralgabe — APPROVED** [review]
thanks for taking this! very nice
will let you revise if you wish and merge if you wish

File: util/helpers_test.go (hunk 1 above)
Comment (singyiu) [reply util/helpers_test.go]: updated

File: util/helpers_test.go (hunk 2 above)
Comment (singyiu) [reply util/helpers_test.go]: Updated.

**Comment by singyiu** [comment]
@decentralgabe 
Thanks a lot for the review.
Updated the nit.
I don't have the permission to merge as a first time contributors to this project though.
Will leave that to you.

**Comment by decentralgabe** [comment]
Thanks @singyiu appreciate it!

**Pull request merged**